vslib/inc/MACsecFilterStateGuard.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "MACsecFilter.h"
+
+namespace saivs
+{
+    class MACsecFilterStateGuard
+    {
+    public:
+        MACsecFilterStateGuard(
+            _Inout_ MACsecFilter::MACsecFilterState &guarded_state,
+            _In_ MACsecFilter::MACsecFilterState target_state);
+
+        ~MACsecFilterStateGuard();
+
+    private:
+        MACsecFilter::MACsecFilterState m_old_state;
+        MACsecFilter::MACsecFilterState &m_guarded_state;
```

Comment:
why this is a reference ?

Reply:
I want to use the RAII mechanism to protect the state at
https://github.com/Azure/sonic-sairedis/blob/cac30e9f4705e88973929a5155a9e76ba23af815/vslib/src/MACsecFilter.cpp#L69
So it's a reference.